### ISSUE TITLE
ResourceExhaustedError (see above for traceback): Ran out of GPU memory when allocating 0 bytes for  	 [[Node: SoftmaxCrossEntropyWithLogits_3 = SoftmaxCrossEntropyWithLogits[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/gpu:3"](Reshape_9, Reshape_10)]]

### ISSUE THREAD

**Issue by SPY-Ming** [SPY-Ming/hmdata#2]
halo :
Have to say that I am not very familiar with github use, so please understand if you are not convenient to read.

I have coded muti-gpu blstm code.It worked before,but recently it has problem:

ResourceExhaustedError (see above for traceback): Ran out of GPU memory when allocating 0 bytes for 
	 [[Node: SoftmaxCrossEntropyWithLogits_3 = SoftmaxCrossEntropyWithLogits[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/gpu:3"](Reshape_9, Reshape_10)]]
	 [[Node: Mean_4/_19597 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:2", send_device_incarnation=1, tensor_name="edge_21859_Mean_4", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

my GPU is tesla m40 and has 11g memory.This error out during training some time and somtimes is out during testing.

I can not find any clue in google and github.So ask help .
my code is flowed:

    def tower():
        costs = []
        accuracys = []
        y_preds = []
        with tf.variable_scope(tf.get_variable_scope()):
            for gpu_id in gpus:
                print 'Build model in gpu %d'%gpu_id
                with tf.device('gpu:%d'%gpu_id):
                    cost,accuracy,y_pred = build_model(gpu_id)
                    tf.get_variable_scope().reuse_variables()
                accuracys.append(accuracy)
                costs.append(cost)
                y_preds.append(y_pred)
        with tf.device('cpu:0'):
            mean_cost = tf.reduce_mean(costs,0)
            mean_accuracy = tf.reduce_mean(accuracys,0)
        return mean_cost,mean_accuracy,y_preds

    def build_model(gpu_id):
        with tf.name_scope('input_dropout'):
            inputs_word = tf.contrib.layers.dropout(word[gpu_id], keep_prob = input_layer_keep_prob)
        with tf.name_scope('blstm_layer'):
            blstm = BLSTM_model(inputs_word,hidden_layer_keep_prob)
            blstm = tf.contrib.layers.dropout(blstm,keep_prob = hidden_layer_keep_prob)
            blstm = tf.reduce_mean(blstm,0)
        outputs= blstm
        weights = tf.get_variable('weight',[2*hidden_size,n_class],
                                  initializer=tf.random_normal_initializer(0, 1))
        biases = tf.get_variable('biases',[n_class],
                                 initializer=tf.random_normal_initializer(0,1))
        weights = tf.contrib.layers.dropout(weights, keep_prob=output_layer_keep_prob)
        
        y_out = tf.matmul(outputs, weights) + biases
        y_pred = tf.argmax(y_out,1)
        regularizers = (tf.nn.l2_loss(weights) + tf.nn.l2_loss(biases) + tf.nn.l2_loss(blstm))
    
        cost = tf.nn.softmax_cross_entropy_with_logits(logits=y_out,labels=tf.cast(label[gpu_id],tf.float32))
        cost += l2_reg_lambda*regularizers
        loss = tf.reduce_mean(cost)
        correct_pred = tf.equal(tf.argmax(y_out,1),tf.argmax(label[gpu_id],1))
        accuracy = tf.reduce_mean(tf.cast(correct_pred,tf.float32))
        return loss,accuracy,y_pred

 cost,accuracy,y_preds =tower()
    print 'Build tower complete!'
    with tf.device('cpu:0'):
        optimizate = tf.train.AdamOptimizer(1e-3).minimize(cost)
    gpu_options =tf.GPUOptions(allow_growth= allow_growth_placement)
    session_conf = tf.ConfigProto(
      allow_soft_placement=allow_soft_placement,
      log_device_placement=log_device_placement,
      gpu_options = gpu_options)
    sess = tf.Session(config=session_conf)
    sess.run(tf.global_variables_initializer())
    start = datetime.datetime.now()
    print 'start time is :',start
    low_acc = [] #for early stop
    low_acc.append(0)
    early_stop = 0
    for epoch in range(1,num_epochs+1):
        every_start = datetime.datetime.now()
        train_cost,train_accuracy = train_model(sess,optimizate,cost,accuracy)
        save_result(os.path.join(result_path,'train_result.txt'), epoch, train_cost,train_accuracy)
        every_end = datetime.datetime.now()
        time = (every_end-every_start).seconds
        print 'epoch:{}\tloss:{}\taccuracy:{}\ttime:{}'.format(epoch,train_cost,train_accuracy,time)
        if epoch % evaluate_every == 0:
            valid_cost,valid_accuracy = valid_model(sess, cost,accuracy)
            save_result(os.path.join(result_path,'valid_result.txt'), epoch, valid_cost,valid_accuracy)
            print 'dataset:sst2\tepoch {}\tloss {}\taccuracy {}'.format(epoch,valid_cost,valid_accuracy)
            low_acc.append(valid_accuracy)
            if valid_accuracy == max(low_acc):
                save_model(sess, model_path+model_name)
            else :
                early_stop += 1
#         if early_stop == 3:
#             print 'early stop,epoch is :%d'%epoch
#             break
    end = datetime.datetime.now()

how can I do ? Thank you very much!
